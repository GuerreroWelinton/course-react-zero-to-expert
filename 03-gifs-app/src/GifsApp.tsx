import { mockGifs } from '@/mock-data/gifs.mock';

import { GifList } from '@/gifs/components/GifList';
import { PreviousSearches } from '@/gifs/components/PreviousSearches';

import { CustomHeader } from '@/shared/components/CustomHeader';
import { SearchBar } from '@/shared/components/SearchBar';
import { useState } from 'react';
import { getGifsByQuery } from './gifs/actions/get-gifs-by-query.action';

export const GifsApp = () => {
  const [previousTerms, setPreviousTerms] = useState(['goku', 'vegeta']);

  const handleTermClicked = (term: string) => {
    console.log({ term });
  };

  const handleSearch = async (query: string = '') => {
    const formattedQuery = query.trim().toLowerCase();

    if (formattedQuery === '') return;

    if (previousTerms.includes(formattedQuery)) return;

    setPreviousTerms([formattedQuery, ...previousTerms].slice(0, 8));

    // setPreviousTerms(prev => {
    //   if (prev.includes(formattedQuery)) return prev;
    //   return [formattedQuery, ...prev].slice(0, 8);
    // });

    const gifs = await getGifsByQuery(query);
    console.log(gifs);
  };

  return (
    <>
      <CustomHeader
        title='Buscador de GIFs'
        description='Descubre y comparte el GIF perfecto'
      />

      <SearchBar placeholder='Busca lo que quieras' onQuery={handleSearch} />

      <PreviousSearches
        searches={previousTerms}
        onLabelClicked={handleTermClicked}
      />

      <GifList gifs={mockGifs} />
    </>
  );
};

// handle => manejar
