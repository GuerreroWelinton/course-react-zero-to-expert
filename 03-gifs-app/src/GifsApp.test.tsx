import { GifsApp } from '@/GifsApp';
import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

describe('GifsApp', () => {
  test('should render component properly', () => {
    const { container } = render(<GifsApp />);

    expect(container).toMatchSnapshot();
  });
});
