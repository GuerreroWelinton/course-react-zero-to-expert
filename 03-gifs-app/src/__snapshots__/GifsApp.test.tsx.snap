// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifsApp > should render component properly 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Buscador de GIFs
    </h1>
    <p>
      Descubre y comparte el GIF perfecto
    </p>
  </div>
  <div
    class="search-container"
  >
    <input
      id="search-input"
      placeholder="Busca lo que quieras"
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Buscar
    </button>
  </div>
  <div
    class="previous-searches"
  >
    <h2>
      BÃºsquedas anteriores
    </h2>
    <ul
      class="previous-searches-list"
    />
  </div>
  <div
    class="gifs-container"
  />
</div>
`;
